File: pulumi-infra-project/pulumi/examples/typescript/digitalocean_firewall_example.ts
// Pulumi DigitalOcean Firewall Example (TypeScript)
import * as pulumi from "@pulumi/pulumi";
import * as digitalocean from "@pulumi/digitalocean";

const firewall = new digitalocean.Firewall("example-firewall", {
    dropletIds: [],
    inboundRules: [{
        protocol: "tcp",
        portRange: "22",
        sourceAddresses: ["0.0.0.0/0"]
    }],
    outboundRules: [{
        protocol: "tcp",
        portRange: "all",
        destinationAddresses: ["0.0.0.0/0"]
    }]
});

export const firewallId = firewall.id;

File: pulumi-infra-project/pulumi/examples/python/digitalocean_firewall_example.py
"""
Pulumi DigitalOcean Firewall Example (Python)

Setup:
  pip install pulumi pulumi_digitalocean
Usage:
  pulumi up
"""
import pulumi
import pulumi_digitalocean as do

firewall = do.Firewall('example-firewall',
    droplet_ids=[],
    inbound_rules=[do.FirewallInboundRuleArgs(
        protocol='tcp',
        port_range='22',
        source_addresses=['0.0.0.0/0']
    )],
    outbound_rules=[do.FirewallOutboundRuleArgs(
        protocol='tcp',
        port_range='all',
        destination_addresses=['0.0.0.0/0']
    )]
)
pulumi.export('firewall_id', firewall.id)
